input logic changed

omaximus opened the pull request.

omaximus committed:
css changed

omaximus committed:
input logic changed

omaximus committed:
comments removed some bugs fixed

vitalyiegorov commented on src/Field.php:
```diff
@@ -309,53 +93,35 @@ public function __save()
-        $previousValue = $this->obj[$this->param];
+        $previousValue = $this->dbObject[$this->param];
+
+        if ($this->param != 'numeric_value' && isset($this->dbObject['numeric_value'])) {
```
@omaximus Зачем нам это, нельзя привязаться к $this->param? У нас может быть и другие поля в таблице, и филд мы можем с  другой таблицей использовать

omaximus replied:
@vitalyiegorov Я сам над этим думал. Это для даты сделано было, я исправлю.
Дата конвертируется в число и записывается в numeric_value.
Конвертирует ф-ия convert() на этапе записи, но обратной конвертации для отображения нет, для это сделать еще 1 ф-ию?

vitalyiegorov commented on src/InputApplication.php:
```diff
@@ -0,0 +1,141 @@
+     */
+    public function createField($entity, $param = null, $identifier = null, $dbQuery = null)
+    {
+        $this->field = new Field($entity, $param, $identifier, $dbQuery);
```
@omaximus Нам же нужно универсально создавать - тоесть каждая App в другом интуте если у него есть свой наследник field должа создавать его єкземпляр, предлагаю сделать protected 
 $fieldClassName = '\samsoncms\input....';
И в наследующих приложениях его переопределять, сдесь в общем коде добавить 
if (class_exists($fieldClassName)) для безопасности и выводить ошибку

vitalyiegorov commented on src/InputApplication.php:
```diff
@@ -0,0 +1,141 @@
+    public function toView($prefix = null, array $restricted = array())
+    {
+        // Generate unique prefix if not passed
+        $innerPrefix = 'field_';
```
@omaximusэто не правильно - нужно свой использовать префикс - $this->fieldClassName, так будет универсально, єто нужно если мы где нибуть когда будет сразу передавать во вьюху два филда а не один что бы одно другое не перетерло, хотя, это внутренняя прорисовка - подумай надо или нет,

omaximus replied:
@vitalyiegorov я думал, в рамках модуля достичь уникальность филда можно разве что их коллекцией.
а уникальность в рамках скрипта зависит от модуля (она есть по идее)?
да и действительно это внутренняя прорисовка, и пока что она работает.

omaximus committed:
New method for fields generation added.
Controller file renamed.
Field rendering moved to Field class
Views were corrected
